src/v2i-hub/RSUHealthMonitorPlugin/src/RSUConfigurationList.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <iostream>
+#include <jsoncpp/json/json.h>
+#include <boost/algorithm/string.hpp>
+#include "RSUConfigurationException.h"
+
+namespace RSUHealthMonitor
+{
+    static constexpr const char *RSUSKey = "RSUS";
+    static constexpr const char *RSUIpKey = "RSUIp";
+    static constexpr const char *SNMPPortKey = "SNMPPort";
+    static constexpr const char *UserKey = "User";
+    static constexpr const char *AuthPassPhraseKey = "AuthPassPhrase";
+    static constexpr const char *RSUMIBVersionKey = "RSUMIBVersion";
+    static constexpr const char *SecurityLevelKey = "SecurityLevel";
+    static constexpr const char *RSU4_1_str = "RSU4.1";
+    static constexpr const char *RSU1218_str = "RSU1218";
+
+    enum class RSUMibVersion
+    {
+        UNKOWN_MIB_V = 0,
+        RSUMIB_V_4_1 = 1,
+        RSUMIB_V_1218 = 2
+    };
+
+    struct RSUConfiguration
+    {
+        std::string rsuIp;
+        uint16_t snmpPort;
+        std::string user;
+        std::string authPassPhrase;
+        std::string securityLevel = "authPriv";
+        RSUMibVersion mibVersion;
+        friend std::ostream &operator<<(std::ostream &os, const RSUConfiguration &config);
+    };
+
+    class RSUConfigurationList
+    {
+    private:
+        std::vector<RSUConfiguration> configs;
+        /***
+         * @brief Parse JSON string and return the corresponding JSON value.
+         * @param rsuConfigsStr  A JSON string includes all RSUs related configrations.
+         * @return JSON::Value A JSON object that includes RSUS information.
+         */
+        Json::Value parseJson(const std::string &rsuConfigsStr) const;
```

Comment:
Why does parse JSON return a JSON value. Isn't the purpose of this object to store the configuration. I would imagine either we have a constructor or initializer that reads in the string and updates the object or we have a function that returns and instance of the object.

```cpp
RSUConfigurationList parse_rsu_config(const std::string &json) {
...
}
...
RSUConfigurationList rsuConfig = parse_rsu_config(jsonString)
```

Not saying this way is inherently wrong, just trying to understand why we return Json value and what your thought process was in setting it up this way.

Reply:
parseJson is a function used by parseRSUS() which is where we store the config objects.